core/string/unicode_normalized_spec.rb

```diff
@@ -0,0 +1,105 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+
+ruby_version_is "2.2" do
+  describe "String#unicode_normalize" do
+    it "raises an Encoding::CompatibilityError if string is not in an unicode encoding" do
+      lambda do
+        "\xE0".force_encoding("ISO-8859-1").unicode_normalize(:nfd)
+      end.should raise_error(Encoding::CompatibilityError)
+    end
+
+    # http://unicode.org/faq/normalization.html#6
+    context "returns normalized form of string by default" do
+      it "03D3 (ϓ) GREEK UPSILON WITH ACUTE AND HOOK SYMBOL" do
+        unicode_03d3 = 'ϓ'
+
+        "\u03D3".unicode_normalize(:nfc).should        == unicode_03d3
+        "\u03D2\u0301".unicode_normalize(:nfd).should  == unicode_03d3
```

Comment:
The assertion should be the other way around. And different nomalization forms are not `==` (or == would become *very* expensive).
```ruby
 "\u03D3".unicode_normalize(:nfd) == "\u03D2\u0301" # => true
```